desktop/renderer/remote-store.desktop.js

```diff
@@ -26,7 +26,7 @@ export default class RemoteStore {
   dispatch (action: any) {
     // TODO use our middlewares
     if (action.constructor === Function) {
-      action(a => this.dispatch(a), () => this.getState())
+      return action(a => this.dispatch(a), () => this.getState())
```

Comment:
We weren't matching what redux thunk does, so you couldn't chain promises that were returned from async actions.

(See this: https://github.com/gaearon/redux-thunk/blob/master/src/index.js) A high contenter for LOC to utility...